src/promptflow-evals/promptflow/evals/synthetic/jailbreak_adversarial_simulator.py

```diff
@@ -0,0 +1,203 @@
+            activity_type=ActivityType.PUBLICAPI,
+            custom_dimensions={
+                "scenario": scenario,
+                "max_conversation_turns": max_conversation_turns,
+                "max_simulation_results": max_simulation_results,
+            },
+        )(func)
+
+        return decorated_func(*args, **kwargs)
+
+    return wrapper
+
+
+class JailbreakAdversarialSimulator:
+    """
+    Initializes the jailbreak adversarial simulator with a project scope.
+
+    :param azure_ai_project: Dictionary defining the scope of the project. It must include the following keys:
+
+        * "subscription_id": Azure subscription ID.
+        * "resource_group_name": Name of the Azure resource group.
+        * "project_name": Name of the Azure Machine Learning workspace.
+        * "credential": Azure credentials object for authentication.
+    :type azure_ai_project: Dict[str, Any]
+    """
+
+    def __init__(self, *, azure_ai_project: Dict[str, Any], credential=None):
+        """Constructor."""
+        # check if azure_ai_project has the keys: subscription_id, resource_group_name, project_name, credential
+        if not all(key in azure_ai_project for key in ["subscription_id", "resource_group_name", "project_name"]):
+            raise ValueError(
+                "azure_ai_project must contain keys: subscription_id, resource_group_name and project_name"
+            )
+        # check the value of the keys in azure_ai_project is not none
+        if not all(azure_ai_project[key] for key in ["subscription_id", "resource_group_name", "project_name"]):
+            raise ValueError("subscription_id, resource_group_name and project_name must not be None")
+        if "credential" not in azure_ai_project and not credential:
+            self.credential = DefaultAzureCredential()
+        elif "credential" in azure_ai_project:
+            self.credential = azure_ai_project["credential"]
+        self.azure_ai_project = azure_ai_project
+        self.token_manager = ManagedIdentityAPITokenManager(
+            token_scope=TokenScope.DEFAULT_AZURE_MANAGEMENT,
+            logger=logging.getLogger("AdversarialSimulator"),
+            credential=self.credential,
+        )
+        self.rai_client = RAIClient(azure_ai_project=azure_ai_project, token_manager=self.token_manager)
+        self.adversarial_template_handler = AdversarialTemplateHandler(
+            azure_ai_project=azure_ai_project, rai_client=self.rai_client
+        )
+
+    def _ensure_service_dependencies(self):
+        if self.rai_client is None:
+            raise ValueError("Simulation options require rai services but ai client is not provided.")
+
+    @monitor_adversarial_scenario
+    async def __call__(
+        self,
+        *,
+        scenario: AdversarialScenario,
+        target: Callable,
+        max_conversation_turns: int = 1,
+        max_simulation_results: int = 3,
+        api_call_retry_limit: int = 3,
```

Comment:
Should we think about them being some kind of retry config param to make it cleaner ?

Reply:
We can try to get something for the task simulator at the design review. I didn't want to change the approved review.